Mathlib/Data/Subtype.lean

```diff
@@ -179,6 +179,14 @@ theorem map_involutive {p : α → Prop} {f : α → α} (h : ∀ a, p a → p (
     (hf : Involutive f) : Involutive (map f h) :=
   fun x ↦ Subtype.ext (hf x)
 
+theorem map_ne {p : α → Prop} {q : β → Prop} {f g : α → β}
+    (h₁ : ∀ a : α, p a → q (f a)) (h₂ : ∀ a : α, p a → q (g a))
+    (h₃ : ∀ a₁ a₂ : α, p a₁ → p a₂ → f a₁ ≠ g a₂)
+    (x y : Subtype p) :
+    map f h₁ x ≠ map g h₂ y := by
```

Comment:
This feels like a very weird condition / conclusion. I had some trouble even figuring out what it said, because I didn't expect it from the name. My Interpretation, which I admit is likely less usable in practice, is
```lean
    (h₃ : Disjoint (f '' {x | p x}) (g '' {x | p x})) :
    Disjoint (Set.range (map f h₁)) (Set.range (map g h₂))
```
Can you add this or something like it to the docstring?